app/profile/page.tsx

```diff
@@ -0,0 +1,299 @@
+
+  const isLoadingUserPositions = userPositions === undefined;
+
+  const spentAmountInUSD = userPositions?.reduce((acc, userPosition) => {
+    const market = new Market(userPosition.fpmm);
+
+    if (market.answer === null) return acc;
+
+    const amountSpentInUSD = tradesCollateralAmountUSDSpent({
+      fpmmTrades: userPosition.fpmmTrades,
+    });
+
+    return amountSpentInUSD + acc;
+  }, 0);
+
+  const winAmountInUSD = userPositions?.reduce((acc, userPosition) => {
+    const position = new Position(userPosition.position);
+    const market = new Market(userPosition.fpmm);
+
+    if (market.answer === null || market.isLoser(position.outcomeIndex - 1)) return acc;
+
+    // missing usd price
+    const winOutcomeBalancesInColleteralToken = tradesOutcomeBalance({
+      fpmmTrades: userPosition.fpmmTrades,
+    });
+
+    return winOutcomeBalancesInColleteralToken + acc;
+  }, 0);
+
+  const potentialWinAmountInUSD = userPositions?.reduce((acc, userPosition) => {
+    const position = new Position(userPosition.position);
+    const market = new Market(userPosition.fpmm);
+
+    if (!userPosition.balance || market.answer !== null || !market.hasLiquidity)
+      return acc;
+
+    const oneShareSellPrice = calcSellAmountInCollateral(
+      userPosition.balance,
+      market.fpmm.outcomeTokenAmounts,
+      position.outcomeIndex - 1,
+      parseFloat(formatEther(market.fpmm.fee))
+    );
+
+    if (!oneShareSellPrice) return acc;
+
+    // missing usd price
+    const priceInColleteralToken = parseFloat(formatEther(oneShareSellPrice));
+
+    return priceInColleteralToken + acc;
+  }, 0);
+
+  const totalVolumeInUSD = userPositions?.reduce(
+    (acc, userPosition) =>
+      tradesVolume({
+        fpmmTrades: userPosition.fpmmTrades,
+      }) + acc,
+    0
+  );
+
+  const numberOfWonBets = userPositions?.reduce((acc, userPosition) => {
+    const position = new Position(userPosition.position);
+    const market = new Market(userPosition.fpmm);
+
+    if (market.answer !== null && market.isWinner(position.outcomeIndex - 1))
+      return acc + 1;
+
+    return acc;
+  }, 0);
+
+  const numberOfClosedBets = userPositions?.reduce((acc, userPosition) => {
+    const market = new Market(userPosition.fpmm);
+
+    if (market.isClosed && market.answer !== null) return acc + 1;
+
+    return acc;
+  }, 0);
+
+  const userFirstParticipationDate = userInfo?.user
+    ? format(fromUnixTime(userInfo.user.firstParticipation), 'MMMM y')
+    : null;
+
+  const positionValue =
+    potentialWinAmountInUSD !== undefined
+      ? formatValueWithFixedDecimals(potentialWinAmountInUSD, 2)
+      : '-';
+  const volumeTraded =
+    totalVolumeInUSD !== undefined
+      ? formatValueWithFixedDecimals(totalVolumeInUSD, 2)
+      : '-';
+  const successRate =
+    numberOfClosedBets !== undefined &&
+    numberOfClosedBets !== 0 &&
+    numberOfWonBets !== undefined
+      ? ((numberOfWonBets / numberOfClosedBets) * 100).toFixed(0)
+      : '-';
```

Comment:
same, this probably could go to the Market entity  

probably will do the "-" here, but the logic to return the successRate.

Reply:
https://github.com/SwaprHQ/presagio/pull/83#discussion_r1745598437 same as here